tests/runtimes/simple-keyvalue/src/methods.rs

```diff
@@ -350,6 +354,94 @@ impl Methods {
             .transpose()
             .map_err(|err| err.to_string())
     }
+
+    /// ElGamal encryption.
+    pub fn encrypt(ctx: &mut TxContext, args: Encrypt) -> Result<Option<Vec<u8>>, String> {
+        if ctx.is_check_only() {
+            return Ok(None);
+        }
+
+        // Derive key pair ID based on the given ID.
+        let hash = Hash::digest_bytes(args.key_pair_id.as_bytes()).0;
+        let key_pair_id = KeyPairId::from(hash.as_ref());
+
+        // Fetch public key.
+        let io_ctx = IoContext::create_child(&ctx.parent.core.io_ctx);
+        let result =
+            ctx.parent
+                .key_manager
+                .get_public_ephemeral_key(io_ctx, key_pair_id, args.epoch);
+        let long_term_pk = tokio::runtime::Handle::current()
+            .block_on(result)
+            .map_err(|err| err.to_string())?
+            .ok_or("public ephemeral key not available")?;
+
+        // Generate ephemeral key. Not secure, but good enough for testing purposes.
```

Comment:
I sure hope no one copy pastes this (or the casual nonce reuse) into actual code.

We have boilerplate code for all of this under runtime/src/common/crypto/mrae, why reinvent the wheel?

Reply:
To use mrae I need a public key and a private key. In my case I want to encrypt a message using public key only. To do that I created an unsafe non-random private key and used it in mrae on line 384. So I think I'm not reinventing anything.

I constructed this unsafe encryption just to test if key manager can generate ephemeral keys. I can probably test this some other way also, e.g. by using signatures (but again the result should be deterministic) or by just returning a simple zero knowledge proof.

Reply:
Ok.